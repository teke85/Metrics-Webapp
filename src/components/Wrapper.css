.page-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 100%;
  height: 180px;
  background-color: #5687e4;
  padding: 10px;
}

.page-image {
  max-width: 50%;
  height: 180px;
  opacity: 0.3;
}

.page-title {
  padding: 2rem;
  max-width: 50%;
  align-items: flex-start;
  text-align: right;
}
