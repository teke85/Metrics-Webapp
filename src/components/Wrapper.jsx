import React from 'react';
import PropTypes from 'prop-types';
import './Wrapper.css';

const PageHeader = (props) => {
  const { map, name, population } = props;
  return (
    <div className="page-header">
      <img src={map} alt={name} className="page-image" />
      <div className="page-title">
        <h2 className="heading-color">{name}</h2>
        <span className="heading-color">{population}</span>
      </div>
    </div>
  );
};
PageHeader.propTypes = {
  name: PropTypes.string,
  population: PropTypes.string,
  map: PropTypes.string,
};

PageHeader.defaultProps = {
  name: '',
  population: '',
  map: '',
};

export default PageHeader;
