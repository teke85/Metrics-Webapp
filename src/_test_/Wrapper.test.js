import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/configureStore';
import '@testing-library/jest-dom';
import Wrapper from '../components/Wrapper';

describe('Wrapper component', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Wrapper />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the PageHeader component with non-default props', () => {
    const { getByAltText, getByText } = render(
      <Wrapper name="Test City" population={100000} map="test-map.png" />,
    );
    const pageImage = getByAltText('Test City');
    const pageTitle = getByText('Test City');
    expect(pageImage).toBeInTheDocument();
    expect(pageTitle).toBeInTheDocument();
  });

  test('renders the correct population in the PageHeader component', () => {
    const { getByText } = render(
      <Wrapper name="Test City" population={100000} map="test-map.png" />,
    );
    const population = getByText('100000');
    expect(population).toBeInTheDocument();
  });
});
