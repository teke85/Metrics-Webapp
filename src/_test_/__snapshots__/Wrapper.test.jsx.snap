// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper component renders correctly 1`] = `
<div
  className="page-header"
>
  <img
    alt=""
    className="page-image"
    src=""
  />
  <div
    className="page-title"
  >
    <h2
      className="heading-color"
    />
    <span
      className="heading-color"
    />
  </div>
</div>
`;
